docs/runtime/runtime-metadata.md

```diff
@@ -0,0 +1,149 @@
+                        " The `AccountId` of the sudo key."
+                    ]
+                }
+            ],
+            "calls":[
+                {
+                    "name":"sudo",
+                    "args":[
+                        {
+                            "name":"proposal",
+                            "type":"Proposal"
+                        }
+                    ],
+                    "documentation":[
+                        " Authenticates the sudo key and dispatches a function call with `Root` origin.",
+                        "",
+                        " The dispatch origin for this call must be _Signed_."
+                    ]
+                },
+                {
+                    "name":"set_key",
+                    "args":[
+                        {
+                            "name":"new",
+                            "type":"Address"
+                        }
+                    ],
+                    "documentation":[
+                        " Authenticates the current sudo key and sets the given AccountId (`new`) as the new sudo key.",
+                        "",
+                        " The dispatch origin for this call must be _Signed_."
+                    ]
+                }
+            ],
+            "events":[
+                {
+                    "name":"Sudid",
+                    "args":[
+                        "bool"
+                    ],
+                    "documentation":[
+                        " A sudo just took place."
+                    ]
+                },
+                {
+                    "name":"KeyChanged",
+                    "args":[
+                        "AccountId"
+                    ],
+                    "documentation":[
+                        " The sudoer just switched identity; the old key is supplied."
+                    ]
+                }
+            ]
+        }
+    ]
+}
+```
+
+As you can see, the metadata of overall runtime is an array of module's metadata. The included modules contain all the dependent SRML and custom runtime modules in your blockchain. 
+
+Now let's walk through each field in module's metadata:
+
+* `name`: the module's name.
+* `prefix`: the name of module's store.
+* `storage`: all the available storage entries in one module, each storage entry is then composed by:
+  * `name`: the storage entry's name.
+  * `modifier`: whether the entry is `Optional` or `Default`.
+  * `type`: the type of the storage entry, could be `PlainType`, `MapType` or `DoubleMapType`.
+  * `documentation`: notes for the entry.
+* `calls`: an array of dispatchable functions:
+  * `name`: the function's name.
+  * `args`: the arguments in function's definition, each includes parameter name and type.
+  * `documentation`: notes for the function.
+* `event`: all the defined events in a module:
+  * `name`: the event's name.
+  * `args`: the arguments in the event definition.
+  * `documentation`: notes for the event.
```

Comment:
What about constants?
```
pub struct ModuleMetadata {
	pub name: DecodeDifferentStr,
	pub prefix: DecodeDifferent<FnEncode<&'static str>, StringBuf>,
	pub storage: ODFnA<StorageEntryMetadata>,
	pub calls: ODFnA<FunctionMetadata>,
	pub event: ODFnA<EventMetadata>,
	pub constants: DFnA<ModuleConstantMetadata>,
}
```

Reply:
Constants didn't exist in 1.0 and from what I understood, this is the documentation for 1.0

Reply:
v1.0 docs live in `website/versioned_docs/version-1.0.0/runtime/`. Since this is `docs/runtime/` it should be up to date with master.

Reply:
I'm all for that! Since 1.0 will (hopefully) be deprecated soon.

Reply:
@joepetrowski My plan is have the content in good quality, then duplicate this content in v1.0, after that I will update the latest doc based on the master branch.